craft_parts/overlays/overlay_manager.py

```diff
@@ -110,3 +117,122 @@ def mkdirs(self) -> None:
             self._project_info.overlay_work_dir,
         ]:
             overlay_dir.mkdir(parents=True, exist_ok=True)
+
+    def refresh_packages_list(self) -> None:
+        """Update the list of available packages in the overlay system."""
+        if not self._overlay_fs:
+            raise RuntimeError("overlay filesystem not mounted")
+
+        mount_dir = self._project_info.overlay_mount_dir
+        with contextlib.suppress(SystemExit), pychroot.Chroot(mount_dir):
+            packages.Repository.refresh_packages_list()
+
+    def download_packages(self, package_names: List[str]) -> None:
+        """Download packages and populate the overlay package cache.
+
+        :param package_names: The list of packages to download.
+        """
+        if not self._overlay_fs:
+            raise RuntimeError("overlay filesystem not mounted")
+
+        mount_dir = self._project_info.overlay_mount_dir
+        with contextlib.suppress(SystemExit), pychroot.Chroot(mount_dir):
+            packages.Repository.download_packages(package_names)
+
+    def install_packages(self, package_names: List[str]) -> None:
+        """Install packages on the overlay area using chroot.
+
+        :param package_names: The list of packages to install.
+        """
+        if not self._overlay_fs:
+            raise RuntimeError("overlay filesystem not mounted")
+
+        mount_dir = self._project_info.overlay_mount_dir
+        with contextlib.suppress(SystemExit), pychroot.Chroot(mount_dir):
+            packages.Repository.install_packages(package_names)
+            shutil.rmtree("/var/cache", ignore_errors=True)
+
+    def fix_resolv_conf(self) -> None:
+        """Work around problems with pychroot when resolv.conf a symlink."""
+        resolv = self._project_info.overlay_mount_dir / "etc" / "resolv.conf"
+        if resolv.is_symlink():
+            resolv.unlink()
+            resolv.touch()
```

Comment:
When it could be a symlink? And a symlink to what? 

Maybe instead of removing it and leaving an empty file in place, would be better to copy the symlinked file?

Reply:
Some images (such as cloudimgs) have this symlink in place:
```
$ ls -l etc/resolv.conf
lrwxrwxrwx 1 root root 39 Mar 11  2021 etc/resolv.conf -> ../run/systemd/resolve/stub-resolv.conf
```
Pychroot needs `resolv.conf` to be a regular file, otherwise it fails to install its own resolver configuration and name resolution will fail inside the chroot.

Reply:
What about copying it instead of leave an empty file?

Reply:
Pychroot only needs a file to bind-mount it's own file on. Our replacement is done on the package cache overlayfs layer, and this layer is discarded after the overlay step execution. The empty file is not migrated or included in the final payload.

Added additional clarification to the source code.